kitsune/users/views.py

```diff
@@ -779,31 +786,43 @@ def get_settings(attr, *args):
             return val
         return import_from_settings(attr, *args)
 
+    def retrieve_matching_jwk(self, header):
+        """Get the signing key by exploring the JWKS endpoint of the OP."""
+        response_jwks = requests.get(
+            self.get_settings('OIDC_OP_JWKS_ENDPOINT'),
+            verify=self.get_settings('OIDC_VERIFY_SSL', True)
+        )
+        response_jwks.raise_for_status()
+        jwks = response_jwks.json()
+
+        key = None
+        for jwk in jwks['keys']:
+            # dummy tokens sent by the fxa-event-broker test script don't have a kid value,
+            # we'll have to see if this is true in dev
+            # if jwk['kid'] != header['kid']:
+            #     continue
+            if 'alg' in jwk and jwk['alg'] != header['alg']:
+                raise SuspiciousOperation('alg values do not match.')
+            key = jwk
+        if key is None:
```

Comment:
you can do `if not key`. This will also catch an empty string for example. I am not sure though if an empty string could be appended to the key though